### PR TITLE
MGMT-4761 Clear all nat iptables rules when tearing down tests using user-managed-networking

### PR DESCRIPTION
**Pull request by ori-amizur**
/cc @tsorya 
/cc @YuviGold

### CODE REVIEW

File: discovery-infra/test_infra/helper_classes/cluster.py [hunk 1]
```diff
@@ -736,14 +737,27 @@ def _configure_load_balancer(self, controller):
...
         lb_controller.set_load_balancing_config(load_balancer_ip, master_ips, worker_ips)
 
+    @classmethod
+    def _get_namespace_index(cls, libvirt_network_if):
+        # Hack to retrieve namespace index - does not exist in tests
+        matcher = re.match(r'^tt(\d+)$', libvirt_network_if)
+        return int(matcher.groups()[0]) if matcher is not None else 0
+
     @classmethod
     def _configure_nat(cls, controller):
```
Comment (eliorerz) [inline discovery-infra/test_infra/helper_classes/cluster.py:747]: Maybe set it as public so we could create fixture on BaseTest

File: discovery-infra/test_infra/helper_classes/cluster.py [hunk 2]
```diff
@@ -736,14 +737,27 @@ def _configure_load_balancer(self, controller):
...
         libvirt_network_if = controller.network_conf.libvirt_network_if
         libvirt_secondary_network_if = controller.network_conf.libvirt_secondary_network_if
         input_interfaces = [libvirt_network_if, libvirt_secondary_network_if]
+        nat_controller = NatController()
+        nat_controller.add_nat_rules(input_interfaces, cls._get_namespace_index(libvirt_network_if))
 
+    @classmethod
+    def _unconfigure_nat(cls, controller):
```
Comment (eliorerz) [inline discovery-infra/test_infra/helper_classes/cluster.py:755]: Same here.. Maybe set it as public so we could create fixture on BaseTest

File: discovery-infra/test_infra/helper_classes/cluster.py (hunk 1 above)
Comment (ori-amizur) [reply discovery-infra/test_infra/helper_classes/cluster.py:747]: OK

File: discovery-infra/test_infra/helper_classes/cluster.py (hunk 2 above)
Comment (ori-amizur) [reply discovery-infra/test_infra/helper_classes/cluster.py:755]: OK

**Pull request merged**